src/main/java/com/hierynomus/smbj/share/File.java

```diff
@@ -87,6 +89,29 @@ public int write(ByteChunkProvider provider, ProgressListener progressListener)
         return writer.write(provider, progressListener);
     }
 
+    /***
+     * Write the data Async in buffer to this file at position fileOffset.
+     * @param buffer     the data to write
+     * @param fileOffset The offset, in bytes, into the file to which the data should be written
+     * @param offset     the start offset in the data
+     * @param length     the number of bytes that are written
+     * @return the Array List of write response future
+     */
+    public ArrayList<Future<SMB2WriteResponse>> writeAsync(byte[] buffer, long fileOffset, int offset, int length) {
```

Comment:
Please make the return type `List<...>` instead of `ArrayList<...>` (and everywhere else in this PR